import_export/static/import_export/action_formats.js

```diff
@@ -1,22 +1,32 @@
-(function($) {
-  $(document).ready(function() {
-    var $actionsSelect, $formatsElement;
-    if ($('body').hasClass('grp-change-list')) {
-        // using grappelli
-        $actionsSelect = $('#grp-changelist-form select[name="action"]');
-        $formatsElement = $('#grp-changelist-form select[name="file_format"]');
-    } else {
-        // using default admin
-        $actionsSelect = $('#changelist-form select[name="action"]');
-        $formatsElement = $('#changelist-form select[name="file_format"]').parent();
-    }
-    $actionsSelect.change(function() {
-      if ($(this).val() === 'export_admin_action') {
-        $formatsElement.show();
-      } else {
-        $formatsElement.hide();
-      }
+(function ($) {
+    $(document).ready(function () {
+        var $actionsSelect, $formatsElement;
+        $actionsSelect = $('.actions select[name="action"]');
+        $formatsElement = $('.actions select[name="file_format"]');
+        if ($('body').hasClass('grp-change-list')) {
+            // using grappelli
+            $ShowElementToShow = $formatsElement;
+        } else {
+            // using default admin
+            $formatsElementToShow = $formatsElement.parent();
+        }
+        for (let i = 0; i < $actionsSelect.length; i++) {
+            $actionsSelect.eq(i).change(function () {
+                if ($(this).val() === 'export_admin_action') {
+                    $formatsElementToShow.eq(i).show();
+                } else {
+                    $formatsElementToShow.eq(i).hide();
+                }
+            });
+            $actionsSelect.eq(i).change();
+
+            $formatsElement.eq(i).change(function () {
+                for (let j = 0; j < $formatsElement.length; j++) {
```

Comment:
I can see that this works to keep the 'format' select boxes in sync, so that if one changes the other is updated to get a new value.  Would it be better to have the handler detect which value was submitted?  So if you click the 'bottom' select with a value of 'csv' it would read 'csv' regardless of what the top dropdown was set to.

We had another ([PR](https://github.com/django-import-export/django-import-export/pull/1219/files)) which detects which menu was clicked (it never got merged because there were no tests).

I think detecting which button was clicked (top or bottom) and then using the value which has been set is cleaner than rewriting the select value.  What do you think?

Reply:
Yes for sure its better to detect which value was submitted, and I'll test it with the work around in the PR you have mentioned.
I just tried to solve it with the fast work around I get in my mind, without need to change in the package python files, as its more simple to override the static file within my project than inherit the class to override the method.